### DISCUSSION TITLE
DGN file is published as BIM file, which is greatly enlarged

### DISCUSSION THREAD

**Discussion by shenshengyi** [iTwin/itwinjs-core#510] [Q&A] [answered]
Hi, my DGN file is only 6m in size, and it is 2G after it is published as BIM file. My DGN file contains many shared geometric elements.

Does BIM not support the sharing of geometric elements, so it becomes so huge after transformation?

**Comment by pmconne** [comment]
iModels do support shared geometry - these are referred to as "geometry parts". How was your iModel produced (what bridge/connector/application was used)?
If you want to share the .bim file, we could take a look to see what is taking up all of that space.

**Reply by shenshengyi** [reply]
> iModels do support shared geometry - these are referred to as "geometry parts". How was your iModel produced (what bridge/connector/application was used)?
> If you want to share the .bim file, we could take a look to see what is taking up all of that space.

Hi, my DGN data is a section of rail, about 4500 sleepers, and the DGN file size is 7m. After using bridge to convert to BIM, the BIM file size is 2G.

Through query, I found that the BIM file contains about 4500 physical object elements related to sleepers, but their IDs are different.

Now I wonder if these sleepers are not shared in BIM, resulting in a sharp increase in file size.

I want to know why the file size has increased dramatically since DGN was transferred to BIM.

**Reply by pmconne** [reply]
The number of physical objects and the fact that their IDs differ does not tell us anything about whether or not they are sharing geometry.
Please answer my previous question (how specifically was this iModel produced - what bridge).
If you can supply the .dgn and/or .bim, even better.

**Comment by pmconne** [comment, marked as answer]
The issue was that the MicroStation bridge would not produce shared geometry parts for shared cells containing stroked line styles. We have changed the code so that it will now do so. The input .dgn file was 7MB; the output .bim file is now 71MB instead of > 2GB.

iModels are always bigger than .dgn files, but they compress well - zipped, this .bim file is 7.7MB.
